src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -236,25 +237,33 @@ void GcsServer::InitGcsJobManager(const GcsInitData &gcs_init_data) {
 
 void GcsServer::InitGcsActorManager(const GcsInitData &gcs_init_data) {
   RAY_CHECK(gcs_table_storage_ && gcs_pub_sub_ && gcs_node_manager_);
-  auto scheduler = std::make_shared<RayletBasedActorScheduler>(
-      main_service_, gcs_table_storage_->ActorTable(), *gcs_node_manager_, gcs_pub_sub_,
-      /*schedule_failure_handler=*/
-      [this](std::shared_ptr<GcsActor> actor) {
-        // When there are no available nodes to schedule the actor the
-        // gcs_actor_scheduler will treat it as failed and invoke this handler. In
-        // this case, the actor manager should schedule the actor once an
-        // eligible node is registered.
-        gcs_actor_manager_->OnActorCreationFailed(std::move(actor));
-      },
-      /*schedule_success_handler=*/
-      [this](std::shared_ptr<GcsActor> actor) {
-        gcs_actor_manager_->OnActorCreationSuccess(std::move(actor));
-      },
-      raylet_client_pool_,
-      /*client_factory=*/
-      [this](const rpc::Address &address) {
-        return std::make_shared<rpc::CoreWorkerClient>(address, client_call_manager_);
-      });
+  std::shared_ptr<GcsActorSchedulerInterface> scheduler;
+  auto schedule_failure_handler = [this](std::shared_ptr<GcsActor> actor) {
+    // When there are no available nodes to schedule the actor the
+    // gcs_actor_scheduler will treat it as failed and invoke this handler. In
+    // this case, the actor manager should schedule the actor once an
+    // eligible node is registered.
+    gcs_actor_manager_->OnActorCreationFailed(std::move(actor));
+  };
+  auto schedule_success_handler = [this](std::shared_ptr<GcsActor> actor) {
+    gcs_actor_manager_->OnActorCreationSuccess(std::move(actor));
+  };
+  auto client_factory = [this](const rpc::Address &address) {
+    return std::make_shared<rpc::CoreWorkerClient>(address, client_call_manager_);
+  };
+
+  if (RayConfig::instance().gcs_actor_scheduling_enabled()) {
+    RAY_CHECK(gcs_resource_manager_ && gcs_resource_scheduler_);
+    scheduler = std::make_shared<GcsBasedActorScheduler>(
```

Comment:
Shuld it be just a unique_ptr?

Reply:
Here is just to work with the shared_ptr in GcsActor.

Reply:
Yes, will fix.